tests/pr_test.go

```diff
@@ -216,3 +219,60 @@ func TestRunUpgradeVpcPattern(t *testing.T) {
 		assert.NotNil(t, output, "Expected some output")
 	}
 }
+
+func TestRunOverride(t *testing.T) {
+	t.Parallel()
+
+	options := setupOptionsQuickStartPattern(t, "slz-ovr", quickStartPatternTerraformDir)
+	options.SkipTestTearDown = true
+	output, err := options.RunTestConsistency()
+
+	if assert.Nil(t, err, "This should not have errored") &&
+		assert.NotNil(t, output, "Expected some output") &&
+		assert.NotNil(t, options.LastTestTerraformOutputs, "Expected some Terraform outputs") {
+		// set override json string with previous value of config output
+		options.TerraformOptions.Vars["override_json_string"] = options.LastTestTerraformOutputs["config"]
+
+		options.SkipTestSetup = true
+		// TERRATEST uses its own internal logger.
+		// The "show" command will produce a very large JSON to stdout which is printed by the logger.
+		// We are temporarily turning the terratest logger OFF (discard) while running "show" to prevent large JSON stdout.
+		options.TerraformOptions.Logger = logger.Discard
+		planStruct, err := terraform.InitAndPlanAndShowWithStructE(options.Testing, options.TerraformOptions)
+		options.TerraformOptions.Logger = logger.Default // turn log back on
+
+		if assert.Nil(t, err, "This should not have errored") &&
+			assert.NotNil(t, planStruct, "Expected some output") {
+			// defines if there is an any resource change (destroy, update, etc)
+			noChange := true
+			for _, resource := range planStruct.ResourceChangesMap {
+				// get JSON string of full changes for the logs
+				changesBytes, changesErr := json.MarshalIndent(resource.Change, "", "  ")
+				// if it errors in the marshall step, just put a placeholder and move on, not important
+				changesJson := "--UNAVAILABLE--"
+				if changesErr == nil {
+					changesJson = string(changesBytes)
+				}
+
+				var resourceDetails string
+				if resource.Change.Actions.Update() {
+					resourceDetails = fmt.Sprintf("Name: %s Address: %s Actions: %s\nDIFF:\n%s\n\nChange Detail:\n%s", resource.Name, resource.Address, resource.Change.Actions, common.GetBeforeAfterDiff(changesJson), changesJson)
+				} else {
+					// Do not print changesJson because might expose secrets
+					resourceDetails = fmt.Sprintf("Name: %s Address: %s Actions: %s\n", resource.Name, resource.Address, resource.Change.Actions)
+				}
+
+				var errorMessage string
+				errorMessage = fmt.Sprintf("Resource(s) identified to be destroyed %s", resourceDetails)
+				noChange := resource.Change.Actions.NoOp() || resource.Change.Actions.Read()
```

Comment:
Two comments here:
1. In golang, the `:=` operator means to assign the value to a new variable in memory, and `=` means to assign value to an existing variable. Since you have a `noChange := true` above, and now using `noChange := ` inside this for loop, this might not be doing what you intended.
2. The logic of `noChange` value might be off here. The `assert` line that follows will not quit the test, only mark it as failed, and then move on with the rest of the `for` loop. So if you want `noChange` to keep track of overall it might not work, since the value will be whatever the last resource was in the loop, and not overall. Probably need one variable to keep track of overall change, and another for the current resource change, and manage so that the `noChange` variable outside the loop indicates that every resource had no change, or "at least one" did have a change.